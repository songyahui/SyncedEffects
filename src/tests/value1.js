"use hopscript"

var hh = require("hiphop");

var prg =
    <hh.module O=${{initValue: 5, combine: (x, y) => x + y}}>
      <hh.loop>
	<hh.emit O value=${5} />
	<hh.emit O value=${10} />
	<hh.pause/>
      </hh.loop>
    </hh.module>

exports.prg = new hh.ReactiveMachine(prg, "value1");
