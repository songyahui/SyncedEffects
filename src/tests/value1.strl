module value1:

output O := 5 : combine integer with +;

loop
	emit O(5);
	emit O(10);
	pause
end loop

end module