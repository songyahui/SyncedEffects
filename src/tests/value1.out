value1> ;
--- Output: O(15)
value1> ;
--- Output: O(15)
value1> ;
--- Output: O(15)
value1> ;
--- Output: O(15)
